power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -45,8 +47,122 @@ struct DatasetInfo {
     Idx batch_size; // for single dataset, the batch size is one
     MetaDataset const* dataset;
     std::vector<ComponentInfo> component_info;
+
+    DatasetInfo(bool is_batch, Idx batch_size, MetaDataset const* dataset)
+        : is_batch(is_batch), batch_size(batch_size), dataset(dataset) {}
 };
 
+template <typename T, dataset_type_tag dataset_type> class ColumnarAttributeRange {
+  public:
+    using Data = std::conditional_t<is_data_mutable_v<dataset_type>, void, void const>;
+
+    class iterator;
+
+    class Proxy {
+      public:
+        using value_type = T;
+
+        Proxy() = default;
+        Proxy(Idx idx, std::span<Data* const> data,
+              std::span<std::reference_wrapper<MetaAttribute const> const> meta_attributes)
+            : idx_{idx}, data_{data}, meta_attributes_{meta_attributes} {
+            assert(data.size() == meta_attributes.size());
+        }
+
+        Proxy& operator=(value_type const& value)
+            requires is_data_mutable_v<dataset_type>
+        {
+            for (Idx attribute_idx = 0; attribute_idx < static_cast<Idx>(meta_attributes_.size()); ++attribute_idx) {
+                auto const& meta_attribute = get_meta_attribute(attribute_idx);
+                std::byte* buffer_ptr = reinterpret_cast<std::byte*>(data_[attribute_idx]) + meta_attribute.size * idx_;
+                meta_attribute.get_value(&value, buffer_ptr, 0);
+            }
+            return *this;
+        }
+        operator value_type() const { return get(); }
+        value_type get() const {
+            std::remove_const_t<value_type> result{};
+            for (Idx attribute_idx = 0; attribute_idx < static_cast<Idx>(meta_attributes_.size()); ++attribute_idx) {
+                auto const& meta_attribute = get_meta_attribute(attribute_idx);
+                std::byte const* buffer_ptr =
+                    reinterpret_cast<std::byte const*>(data_[attribute_idx]) + meta_attribute.size * idx_;
+                meta_attribute.set_value(&result, buffer_ptr, 0);
```

Comment:
Same here
This will go through a function pointer call every time. As this is a very tight loop, we have another function which does not go to function pointer:

https://github.com/PowerGridModel/power-grid-model/blob/e22b528cc8084d45ef86c7140fe97e8aa0fb5e45/power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/meta_data.hpp#L102-L107

Reply:
See example here

https://github.com/PowerGridModel/power-grid-model/blob/e22b528cc8084d45ef86c7140fe97e8aa0fb5e45/power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/serializer.hpp#L504-L509

Reply:
Done